src/components/ServiceList.jsx

```diff
@@ -1,22 +1,67 @@
 import { Alert } from "@baltimorecounty/dotgov-components";
 import FilterList from "../components/FilterList";
-import { MostPopularServiceIconStyles } from "../styles";
-import React from "react";
-import ServiceIconLink from "../components/ServiceIconLink";
+import React, { useState } from "react";
+import ServiceIconLink from "./ServiceIconLink";
+import SearchCollapse from "./SearchCollapse";
 import useServices from "../hooks/useServices";
+import {
+  MostPopularServiceIconStyles,
+  LegendText,
+  CollapseContainer
+} from "../styles";
+import Search from "./Search";
 
 const ServiceList = () => {
   const { hasError, serviceItems = [], isLoading } = useServices();
+  const [isChecked, setChecked] = useState(false);
+  const [searchedItems, setSearchedItems] = useState([]);
+  const [searchText, setSearchText] = useState([]);
+  const [isFiltering, setIsFiltering] = useState(false);
 
-  let legendText = {
-    fontStyle: "italic",
-    marginLeft: "5px",
-    fontWeight: "900"
+  const filterItems = (services, searchText) => {
+    return services.filter(item => {
+      const { name, department } = item;
+      return (
+        name.toLowerCase().indexOf(searchText.toLowerCase()) !== -1 ||
+        department.toLowerCase().indexOf(searchText.toLowerCase()) !== -1
+      );
+    });
   };
 
-  let flex = {
-    display: "flex",
-    flexDirection: "row"
+  const filterByPopularity = item => {
+    return item.rank > 0 ? true : false;
+  };
+
+  const checkCondition = (checkedVal, searchText) => {
+    let checkedItems = [];
+    const searchTextLength = searchText.length > 0;
+    checkedItems = checkedVal
+      ? searchTextLength
+        ? filterItems(serviceItems.filter(filterByPopularity), searchText)
+        : serviceItems.filter(filterByPopularity)
+      : searchTextLength
+      ? filterItems(serviceItems, searchText)
+      : [];
+
+    setSearchedItems(checkedItems);
+  };
+  const settingFiltering = (searchText, checkedValue) => {
+    setIsFiltering(searchText === 0 && checkedValue === false ? false : true);
```

Comment:
this tenary can be simplified. in this case you would just need to do the opposite of this condition.

```js
setIfFiltering(searchText || checkedValue);
```
**note** i think the logic above is right, but is not tested


this probably removes the need for `settingFiltering`, you can probably just call `setIsFiltering` in place of it

Reply:
slight modification but end result is still removal of the extra function call.